### PR TITLE
Argparse improvements

### PR DESCRIPTION
**Pull request by mlech-reef**
```
b2 [-h]
   {authorize-account,cancel-all-unfinished-large-files,cancel-large-file,clear-account,copy-file-by-id,create-bucket,create-key,delete-bucket,delete-file-version,delete-key,download-file-by-id,download-file-by-name,get-account-info,get-bucket,get-file-info,get-download-auth,get-download-url-with-auth,hide-file,list-buckets,list-file-versions,list-file-names,list-keys,list-parts,list-unfinished-large-files,ls,make-url,make-friendly-url,sync,update-bucket,upload-file,version}
   ...

  This program provides command-line access to the B2 service.

  The environment variable B2_ACCOUNT_INFO specifies the sqlite
  file to use for caching authentication information.
  The default file to use is: ~/.b2_account_info

  For more details on one command: b2 <command> --help

  When authorizing with application keys, this tool requires that the key
  have the 'listBuckets' capability so that it can take the bucket names
  you provide on the command line and translate them into bucket IDs for the
  B2 Storage service.  Each different command may required additional
  capabilities.  You can find the details for each command in the help for
  that command.

optional arguments:
  -h, --help  show this help message and exit

usages:
  b2 authorize-account [-h] [applicationKeyId] [applicationKey]
  b2 cancel-all-unfinished-large-files [-h] bucketName
  b2 cancel-large-file [-h] filed
  b2 clear-account [-h]
  b2 copy-file-by-id [-h] [--metadataDirective {copy,replace}]
                     [--contentType CONTENTTYPE] [--range RANGE] [--info INFO]
                     sourceFileId destinationBucketName b2FileName
  b2 create-bucket [-h] [--bucketInfo BUCKETINFO] [--corsRules CORSRULES]
                   [--lifecycleRules LIFECYCLERULES]
                   bucketName bucketType
  b2 create-key [-h] [--bucket BUCKET] [--namePrefix NAMEPREFIX]
                [--duration DURATION]
                keyName capabilities
  b2 delete-bucket [-h] bucketName
  b2 delete-file-version [-h] [fileName] filed
  b2 delete-key [-h] applicationKeyId
  b2 download-file-by-id [-h] [--noProgress] fileId localFileName
  b2 download-file-by-name [-h] [--noProgress]
                           bucketName b2FileName localFileName
  b2 get-account-info [-h]
  b2 get-bucket [-h] [--showSize] bucketName
  b2 get-file-info [-h] filed
  b2 get-download-auth [-h] [--prefix PREFIX] [--duration DURATION] bucketName
  b2 get-download-url-with-auth [-h] [--duration DURATION] bucketName fileName
  b2 hide-file [-h] bucketName fileName
  b2 list-buckets [-h]
  b2 list-file-versions [-h]
                        bucketName [startFileName] [startFileId] [maxToShow]
                        [prefix]
  b2 list-file-names [-h] bucketName [startFileName] [maxToShow] [prefix]
  b2 list-keys [-h] [--long]
  b2 list-parts [-h] largeFileId
  b2 list-unfinished-large-files [-h] bucketName
  b2 ls [-h] [--long] [--versions] [--recursive] [--prefix]
        bucketName [folderName]
  b2 make-url [-h] filed
  b2 make-friendly-url [-h] bucketName fileName
  b2 sync [-h] [--noProgress] [--dryRun] [--allowEmptySource]
          [--excludeAllSymlinks] [--threads THREADS]
          [--compareVersions {none,modTime,size}] [--compareThreshold MILLIS]
          [--excludeRegex REGEX] [--includeRegex REGEX]
          [--excludeDirRegex REGEX] [--excludeIfModifiedAfter TIMESTAMP]
          [--skipNewer | --replaceNewer] [--delete | --keepDays DAYS]
          source destination
  b2 update-bucket [-h] [--bucketInfo BUCKETINFO] [--corsRules CORSRULES]
                   [--lifecycleRules LIFECYCLERULES]
                   bucketName bucketType
  b2 upload-file [-h] [--noProgress] [--quiet] [--contentType CONTENTTYPE]
                 [--minPartSize MINPARTSIZE] [--sha1 SHA1] [--threads THREADS]
                 [--info INFO]
                 bucketName localFilePath b2FileName
  b2 version [-h]
```
```
b2 authorize-account [-h] [applicationKeyId] [applicationKey]

  Prompts for Backblaze applicationKeyId and applicationKey (unless they are given
  on the command line).

  You can authorize with either the master application key or
  a normal application key.

  To use the master application key, provide the application key ID and
  application key from the "B2 Cloud Storage Buckets" page on
  the web site: https://secure.backblaze.com/b2_buckets.htm

  To use a normal application key, created with the create-key
  command or on the web site, provide the application key ID
  and the application key itself.

  You can also optionally provide application key ID and application key
  using environment variables B2_APPLICATION_KEY_ID and
  B2_APPLICATION_KEY respectively.

  Stores an account auth token in ~/.b2_account_info by default,
  or the file specified by the B2_ACCOUNT_INFO environment variable.

  Requires capability: listBuckets

positional arguments:
  applicationKeyId
  applicationKey

optional arguments:
  -h, --help        show this help message and exit
```

### CODE REVIEW

**Comment by ppolewicz** [comment]
b2 -h is very sparse, it used to not have empty lines between commands and I think that was better

**Comment by mlech-reef** [comment]
> b2 -h is very sparse, it used to not have empty lines between commands and I think that was better

Agree. I have corrected it.

File: b2/console_tool.py [hunk 1]
```diff
@@ -1252,9 +1321,8 @@ def run(self, args):
 
     def get_policies_manager_from_args(self, args):
         if args.includeRegex and not args.excludeRegex:
-            raise CommandError(
-                '--includeRegex cannot be used without --excludeRegex at the same time'
-            )
+            logger.warning('--includeRegex is skipped without --excludeRegex')
```
Comment (ppolewicz) [inline b2/console_tool.py:1324]: it has to be an error

File: b2/console_tool.py [hunk 2]
```diff
@@ -1119,7 +1178,7 @@ class Sync(Command):
     is a regular expression that is tested against the full path
     of each file.
 
-    Note that --includeRegex cannot be used without --excludeRegex.
+    Note that --includeRegex is skipped without --excludeRegex.
```
Comment (ppolewicz) [inline b2/console_tool.py:1181]: it has to be an error

File: b2/console_tool.py [hunk 3]
```diff
@@ -137,30 +118,64 @@ class Command(object):
...
-            description=cls._format_description(cls.__doc__),
-            formatter_class=argparse.RawTextHelpFormatter,
-            parents=[parent]
-        )
-        cls._setup_subparser(parser)
-        return parser
+    def register_subcommand(cls, command_class):
+        assert cls.subcommands_registry is not None, 'Initialize the registry class'
+        return cls.subcommands_registry.register(key=mixed_case_to_hyphens(command_class.__name__)
+                                                )(command_class)
```
Comment (ppolewicz) [inline b2/console_tool.py:134]: please reformat it somehow to make it more readable, it's super weird now

File: b2/console_tool.py (hunk 1 above)
Comment (mlech-reef) [reply b2/console_tool.py:1324]: As discussed with @ppolewicz , I'm going to implement this assertion in SDK, not CLI.

File: b2/console_tool.py (hunk 2 above)
Comment (mlech-reef) [reply b2/console_tool.py:1181]: As discussed with @ppolewicz , I'm going to implement this assertion in SDK, not CLI.

Comment (mlech-reef) [reply b2/console_tool.py]: I've reverted this line.

File: b2/console_tool.py (hunk 1 above)
Comment (mlech-reef) [reply b2/console_tool.py]: I've removed this log completely

File: b2/console_tool.py (hunk 3 above)
Comment (mlech-reef) [reply b2/console_tool.py]: Fixed

**Pull request merged**